share/share.go

```diff
@@ -40,6 +43,31 @@ func GetData(s Share) []byte {
 	return s[NamespaceSize:]
 }
 
+// ShareWithProof contains data with corresponding Merkle Proof
+type ShareWithProof struct { //nolint: revive
```

Comment:
Let's name it sample

Reply:
ShareWithProof:
- It is not always sample. The type is also used within BEFP pkg, where `sample` name will be confusing and might be used in other implementation in future. It should be named protocol agnostic. 
- it literally extends `Share` type with Proof, so named according to the content inside